server/state/index.js

```diff
@@ -10,8 +10,12 @@ const SmartContractsStore = () => {
 
     unsubscribe: address => {
       const contractIndex = smartContracts.findIndex(i => i.address === address)
-      smartContracts.get(contractIndex).listener.unsubscribe()
-      smartContracts = smartContracts.delete(contractIndex)
+      const contractToRemove = smartContracts.get(contractIndex)
+
+      if (contractToRemove) {
+        if (contractToRemove.listener) contractToRemove.listener.unsubscribe()
```

Comment:
When would be a time that we have the smart contract in state, but it doesn't have a listener?

Reply:
testing

Reply:
basically demoSmartContractJSON doesn't have listener field, but the smart contract object that is added via handleListenEvent does